iceoryx_examples/ice_multi_publisher/README.md

```diff
@@ -0,0 +1,262 @@
+
+
+    iox::capro::IdString instance{iox::cxx::TruncateToCapacity, instanceName};
+    iox::popo::TypedPublisher<CounterTopic> publisher({"Group", instance, "Counter"});
+
+
+After construction, we immediately offer the topic and start sending data.
+
+    publisher.offer();
+
+    for (uint32_t counter = 0U; !killswitch; ++counter)
+    {
+        CounterTopic data{counter, id};
+        publisher.publishCopyOf(data);
+        //...
+    }
+
+The data consists of an id (chosen to be different for each publisher to distinguish their data) and a monotonically increasing counter. This counter is send periodically and we leave the loop when Ctrl-C is pressed and stop offering.
+
+    publisher.stopOffer();
+
+
+In the main function two threads are started, each of them corresponding to a publisher.
+Notice that they have different ids but use the same ``"Instance"``. While the instance could have been a string literal in the publisher constructor, this way it is possible to choose the instance name at runtime. This could be done for the group and topic identifier as well. Finally both publishers send at different time intervals, roughly 500ms and 1000ms.
+
+    std::thread sender1(send, 1, "Instance", std::chrono::milliseconds(500));
+    std::thread sender2(send, 2, "Instance", std::chrono::milliseconds(1000));
+
+Once Ctrl+C is pressed, we leave the publisher loops and join the threads.
+
+    sender1.join();
+    sender2.join();
+
+### Subscriber application
+
+We create a subscriber via
+    iox::popo::TypedSubscriber<CounterTopic> subscriber({"Group", "Instance", "Counter"});
+
+and immediately subscribe.
+    subscriber.subscribe();
+
+Notice that all identifiers match the ones provided by the two publishers.
+
+We periodically wake up
+
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    while (subscriber.hasNewSamples())
+
+When there are new samples we display them on the console.
+
+    subscriber.take()
+        .and_then([](iox::popo::Sample<const CounterTopic>& sample) {
+            std::cout << "Received: " << *sample.get() << std::endl;
+        })
+
+The topic is displayed by providing a defintion of ``operator<<`` which prints the id and counter to the console.
+The displayed counters are monotonically increasing for each id but between different publishers the order of data arrival is indeterminate due to concurrent sending.
+
+We also handle potential errors
+
+    .or_else([](iox::popo::ChunkReceiveError) { std::cout << "Error while receiving." << std::endl; });
+
+and wait for some time before looking for data again.
+
+    std::cout << "Waiting for data ... " << std::endl;
+
+When Ctrl+C is pressed we exit the loop and unsubscribe
+
+    subscriber.unsubscribe();
+
+before joining the receiver thread
+
+    receiver.join();
+
+### Resubscriber application
+
+We first define a bound for the maximum number of samples the subscriber can hold.
```

Comment:
the subscriber can queue up. How many it can hold is for my how many samples the user can take and keep in parallel. This is another parameter that is currently compile time